pkg/docker/config/config.go

```diff
@@ -257,17 +257,15 @@ func modifyJSON(sys *types.SystemContext, editor func(auths *dockerConfigFile) (
 	if err != nil {
 		return err
 	}
+	if legacyFormat {
+		return fmt.Errorf("writes to %s using legacy format are not supported", path)
+	}
 
 	dir := filepath.Dir(path)
-	if _, err := os.Stat(dir); os.IsNotExist(err) {
-		if err = os.MkdirAll(dir, 0700); err != nil {
-			return errors.Wrapf(err, "error creating directory %q", dir)
-		}
+	if err = os.MkdirAll(dir, 0700); err != nil {
+		return err
```

Comment:
I'd prefer the old error  message.
```
return errors.Wrapf(err, "error creating directory %q", dir)
```

Reply:
so, `os.MkdirAll` already has the operation (`mkdir`) and the directory name in its error message (see `os.PathError`), so there's no need to wrap it unless you want to add some higher-level context, which does not seem to be the case here, that is why I (semi-automatically) removed it here.

Reply:
I don't think we need to wrap.

This test will give
```
err := os.MkdirAll("/test/foo/bar", 0700)
fmt.Println(err)
```
Gives an error like:
```
go run test.go 
mkdir /test: permission denied
```

Reply:
...and forgot to describe it in commit message. Fixed.